src/deepsparse/transformers/helpers.py

```diff
@@ -193,3 +199,93 @@ def _normalize_fields(_dict):
+    :param emb_extraction_layer: last bert layer to include.
+        default -1 (last layer)
+    :param final_node_name: name of last computed node in graph
+    :param output_name: name of graph output, default "embedding"
+    :param output_path: path to write resulting onnx file. If not provided,
+        will create a temporary file path that will be destroyed on program end
+    :return: if no output path, a tuple of the saved path to the model, list of
+        model output names, and reference to the tempfile object will be returned
+        otherwise, only the model output names will be returned
+    """
+
+    def _check_initializer_names(model: ModelProto) -> Union[str, None]:
+        bert_layer_prog = re.compile(
+            r"bert\.encoder\.layer\.\d+\.output\.LayerNorm\.bias"
+        )
+        distillbert_layer_prog = re.compile(
+            r"distilbert.transformer.layer.\d+\.output_layer_norm.bias"
+        )
+
+        layer_init_names = [
+            initializer.name
+            for initializer in model.graph.initializer
+            if bert_layer_prog.match(initializer.name)
+            or distillbert_layer_prog.match(initializer.name)
+        ]
+        assert len(layer_init_names) > 0
+
+        final_node_initializer_name = sorted(
+            layer_init_names,
+            key=lambda name: int(re.findall("[0-9]+", name)[0]),
+            reverse=False,
+        )[emb_extraction_layer]
+
+        return [
+            node.name
+            for node in model.graph.node
+            if final_node_initializer_name in node.input
+        ][0]
+
+    # Determine where to cut the model
+    if final_node_name is None:
+        model = onnx.load(model_path)
+        final_node_name = None
+
+        # Try to match bert layers by initializer names
+        try:
+            final_node_name = _check_initializer_names(model)
+        except Exception as exception:
+            _LOGGER.info(f"Failed to truncate transformer: {exception}")
+
+        if final_node_name is None:
+            raise Exception("Failed to truncate transformer model")
+
+    # Override outputs to create subgraph
+    _LOGGER.info(f"Truncating transformer model to {final_node_name}")
+    if output_path is None:
+        tmp_file = NamedTemporaryFile()  # file will be deleted after program exit
+
+        truncate_onnx_model(
+            onnx_filepath=model_path,
+            output_filepath=tmp_file.name,
+            final_node_names=[final_node_name],
+            graph_output_names=[output_name],
+            graph_output_shapes=[[None, model_size]],
+        )
+        return tmp_file.name, [output_name], tmp_file
+    else:
+        truncate_onnx_model(
+            onnx_filepath=model_path,
+            output_filepath=output_path,
+            final_node_names=[final_node_name],
+            graph_output_names=[output_name],
+            graph_output_shapes=[[None, model_size]],
+        )
+        return [output_name]
```

Comment:
to keep results the same, any reason not to return `None, x, None`? what do you see as the intended use here?

Reply:
This function is meant to match the schema of overwrite_transformer_onnx_model_inputs. I imagine the reasoning being that it makes it absolutely clear to the caller that they should handle the temp file case separately, but I'm not attached to this paradigm.

Reply:
Decided to return 3 tuple in both cases